k2/csrc/context_inl.cuh

```diff
@@ -0,0 +1,182 @@
+/**
+ * @brief
+ * context_inl
+ * this file is supposed to be included only by context.h
+ * It' the implementation of template functions and inlined functions.
+ *
+ * @copyright
+ * Copyright (c)  2020  Xiaomi Corporation (authors: Daniel Povey
+ *                                                   Meixu Song)
+ *
+ * @copyright
+ * See LICENSE for clarification regarding multiple authors
+ */
+
+#ifndef K2_CSRC_CONTEXT_INL_H_
+#define K2_CSRC_CONTEXT_INL_H_
+
+#ifndef IS_IN_K2_CSRC_CONTEXT_H_
+#error "this file is supposed to be included only by context.h"
+#endif
+
+// No header guard for this file since it will only be included
+// in ops.h
```

Comment:
inlcuded in `op.h`? I guess it's `context.h`?

Reply:
delete those two lines as it's with guard now. And to your question, files: ops.* are changed to array_ops.*, as dan required.